### PR TITLE
Fall back to an RGBA4444 canvas configuration if 8888 is not supported

### PR DESCRIPTION
**Pull request by jason-simmons**

### CODE REVIEW

File: sky/shell/gpu/direct/ganesh_canvas.cc [hunk 1]
```diff
@@ -51,7 +67,7 @@ SkCanvas* GaneshCanvas::GetCanvas(int32_t fbo, const SkISize& size) {
   GrBackendRenderTargetDesc desc;
   desc.fWidth = size.width();
   desc.fHeight = size.height();
-  desc.fConfig = kSkia8888_GrPixelConfig;
+  DCHECK(SelectPixelConfig(&desc.fConfig));
```
Comment (abarth) [inline sky/shell/gpu/direct/ganesh_canvas.cc]: `DCHECK` should probably be `CHECK`.  The argument to `DCHECK` isn't evaluated in release builds.

**Comment by abarth** [comment]
LGTM

File: sky/shell/gpu/direct/ganesh_canvas.cc (hunk 1 above)
Comment (jason-simmons) [inline sky/shell/gpu/direct/ganesh_canvas.cc]: done

**Comment by j9brown** [comment]
12 bit color, woot!  ;)

(Does this actually perform well on current hardware?)

**Comment by jason-simmons** [comment]
This was added to support some Android emulator configurations where RGBA8888 was not available.  I haven't seen any need for this on actual hardware.

**Pull request merged**